pinery-miso/src/main/java/ca/on/oicr/pinery/lims/miso/converters/SampleTypeConverter.java

```diff
@@ -0,0 +1,132 @@
+      this.libraryType = libraryType;
+      this.dilutionType = dilutionType;
+    }
+
+    public String getLibraryType() {
+      return libraryType;
+    }
+
+    public String getDilutionType() {
+      return dilutionType;
+    }
+  }
+
+  private static final String SAMPLE_TYPE_UNKNOWN = "Unknown";
+
+  /**
+   * Translates a MISO Sample Class to a Pinery Sample Type
+   * 
+   * @param misoType SampleClass alias from MISO
+   * @return the corresponding Pinery Sample Type String
+   */
+  public static String getSampleType(String misoSampleClass) {
+    if (misoSampleClass == null) {
+      return "Sample";
+    }
+    return misoSampleClass
+        .replace(" (stock)", "")
+        .replace(" (aliquot)", "")
+        .replace("LCM Tube", "LCM Tubes");
+  }
+
+  /**
+   * Determines the correct Pinery Sample Type to assign to a MISO Library of Library Dilution
+   * 
+   * @param misoType "Library" or "Dilution"
+   * @param platformName MISO LibraryType platformType
+   * @param libraryType MISO LibraryType description
+   * @return The Pinery SampleType String that corresponds to the parameters given, or "Unknown" if it cannot be determined
+   */
+  public static String getNonSampleSampleType(String misoType, String platformName, String libraryType) {
+    if (platformName == null) {
+      log.debug("Cannot determine SampleType due to null platformName");
+      return SAMPLE_TYPE_UNKNOWN;
+    }
+
+    switch (platformName) {
+    case PLATFORM_ILLUMINA:
+      IlluminaSampleType sType = null;
+      if (libraryType == null) {
+        log.debug("Cannot determine SampleType due to null libraryType");
+        return SAMPLE_TYPE_UNKNOWN;
+      }
+
+      switch (libraryType) {
+      case LIBRARY_TYPE_MRNA:
+        sType = IlluminaSampleType.M_RNA;
+        break;
+      case LIBRARY_TYPE_PAIRED_END:
+        sType = IlluminaSampleType.PE;
+        break;
+      case LIBRARY_TYPE_SMALL_RNA:
+        sType = IlluminaSampleType.SM_RNA;
+        break;
+      case LIBRARY_TYPE_SINGLE_END:
+        sType = IlluminaSampleType.SE;
+        break;
+      case LIBRARY_TYPE_WHOLE_TRANSCRIPTOME:
+        sType = IlluminaSampleType.WT;
+        break;
+      case LIBRARY_TYPE_MATE_PAIR:
+        sType = IlluminaSampleType.MP;
+        break;
+      case LIBRARY_TYPE_TOTAL_RNA:
+        sType = IlluminaSampleType.TOTAL_RNA;
+        break;
+      default:
+        log.debug("Unexpected LibraryType: " + libraryType + ", Cannot determine Sample Type");
+        return SAMPLE_TYPE_UNKNOWN;
+      }
+
+      switch (misoType) {
+      case MISO_TYPE_LIBRARY:
+        return sType.getLibraryType();
+      case MISO_TYPE_DILUTION:
+        return sType.getDilutionType();
+      default:
+        log.debug("Unexpected MISO type: " + misoType + ". Cannot determine Sample Type");
+        return SAMPLE_TYPE_UNKNOWN;
+      }
+    default:
+      log.debug("Unknown platform: " + platformName + ". Cannot determine Sample Type");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use the built-in formatting to construct this argument. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2629)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Format specifiers should be used instead of string concatenation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3457)